pkg/provider/identity.go

```diff
@@ -35,26 +54,235 @@ type PeerIdentity struct {
 type FullIdentity struct {
 	PeerIdentity
 	PrivateKey crypto.PrivateKey
+}
 
-	todoCert *tls.Certificate // TODO(jt): get rid of this and only use the above
+type CAConfig struct {
+	CertPath   string `help:"path to the certificate chain for this identity" default:"$CONFDIR/identity.leaf.cert"`
+	KeyPath    string `help:"path to the private key for this identity" default:"$CONFDIR/identity.leaf.key"`
+	Difficulty uint16 `default:"24" help:"minimum difficulty for identity generation"`
+	Version    string `help:"semantic version of CA storage format"`
```

Comment:
version seems like it's related to loading, but not to the in memory representation, right? after you load a CA the version doesn't matter anymore?